app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::StateMachine
+  def customize_destination
```

Comment:
@durandom 
Yes, that is why I decided to override this method. It is my understanding that Azure is the only cloud provider where an IP addresses is a **required** property for VM creation. In the other cloud providers is not, instead it is set during post provisioning. While Azure remains the exception I thought overriding the method was the least disruptive approach.